src/test/scripts/check_enterprise_merge.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# fail if trying to reference a variable that is not set.
+set -u
+# exit immediately if a command fails
+set -e
+
+# try_merge sees if we can merge "from" branch to "to" branch
+# it will exit with nonzero code if the merge fails because of conflicts.
+try_merge() {
+    to=$1
+    from=$2
+    git checkout "${to}"
+    # this will exit since -e option is set and it will return non-zero code on conflicts.
+    git merge --no-ff --no-commit "${from}"
+
+}
+
+git config --global user.email "citus-bot@microsoft.com" 
+git config --global user.name "citus bot" 
+
+cd ~
+git clone https://${GIT_USERNAME}:${GIT_TOKEN}@github.com/citusdata/citus-enterprise
+cd citus-enterprise
+git fetch --all
+
+# echo commands
+set -x
+git branch -r --list
+
+branch_name="${CIRCLE_BRANCH}"
+
+if [ `git branch -r --list origin/$branch_name` ]
+then 
+    try_merge enterprise-master origin/$branch_name
```

Comment:
What happens if the same branch name exists in enterprise and community?

Reply:
If you open a branch in community say feature/X and that branch was already in enterprise but it was something different, then this job won't detect conflicts. I think it is a low chance and in that case with some other PR we will realize that there were some conflicts because of feature/X. 

Or say that feature/X on community would create a conflict on enterprise, it is possible that someone will create feature/X on enterprise but the content of feature/X on enterprise will be different than community feature/X. In that case again the job could still pass. So currently the author is responsible for making sure that feature/X is indeed what we want on enterprise.